fs/vfs/fs_read.c

```diff
@@ -50,19 +50,14 @@
  *
  ****************************************************************************/
 
-static ssize_t file_readv_compat(FAR struct file *filep, FAR struct uio *uio)
+static ssize_t file_readv_compat(FAR struct file *filep,
+                                 FAR const struct iovec *iov, int iovcnt)
 {
-  FAR const struct iovec *iov = uio->uio_iov;
-  int iovcnt = uio->uio_iovcnt;
   FAR struct inode *inode = filep->f_inode;
   ssize_t ntotal;
   ssize_t nread;
-  size_t remaining;
-  FAR uint8_t *buffer;
   int i;
 
-  DEBUGASSERT(inode->u.i_ops->read != NULL);
```

Comment:
how is this assertion removal related to the rest of this commit? ditto for writev

Reply:
already have the check before call this internal function:

![image](https://github.com/user-attachments/assets/79224e39-1384-4829-999a-e2f44b245e74)

Reply:
yes, this assertion is to ensure the condition.
and "the assertion is currently always true" is not a good reason to remove an assertion, IMO.

Reply:
Because this is an internal function, and the calling site can ensure that the assertion condition is not true, we should reduce invalid assertions as much as possible.

Reply:
i disagree. nothing wrong with having assertions like this, especially for complex functions like this.
but i understood your opinion.